mockgen/model/model.go

```diff
@@ -343,6 +343,13 @@ func typeFromType(t reflect.Type) (Type, error) {
 	}
 
 	if imp := t.PkgPath(); imp != "" {
+		// PkgPath might return a path that includes "vendor"
+		// These paths do not compile, so we need to remove everything
+		// up to and including "/vendor/"
+		// see https://github.com/golang/go/issues/12019
+		if i := strings.LastIndex(imp, "/vendor/"); i != -1 {
```

Comment:
I'm not familiar with the `internal/` directory problem--could you elaborate? Is it necessary to address that in this PR?

I did some searches and didn't find a library for this. IMO it's such a small piece of logic a library feels like overkill.